src/EventGrid/EventGrid/Domain/NewAzureEventGridDomainKey.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    [Cmdlet(
+        VerbsCommon.New,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomainKey",
+        DefaultParameterSetName = DomainNameParameterSet,
+        SupportsShouldProcess = true), 
+    OutputType(typeof(DomainSharedAccessKeys))]
+
+    public class NewAzureEventGridDomainKey : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            ParameterSetName = DomainNameParameterSet,
+            HelpMessage = EventGridConstants.ResourceGroupNameHelp)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasResourceGroup)]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            ParameterSetName = DomainNameParameterSet,
+            HelpMessage = EventGridConstants.DomainNameHelp)]
+        [ResourceNameCompleter("Microsoft.EventGrid/domains", nameof(ResourceGroupName))]
+        [ValidateNotNullOrEmpty]
+        public string DomainName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2,
+            ParameterSetName = DomainNameParameterSet,
+            HelpMessage = EventGridConstants.KeyNameHelp)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.KeyNameHelp,
+            ParameterSetName = DomainInputObjectParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1,
+            HelpMessage = EventGridConstants.KeyNameHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainInputObjectHelp,
+            ParameterSetName = DomainInputObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSDomain InputObject { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = EventGridConstants.DomainResourceIdHelp,
+            ParameterSetName = ResourceIdEventSubscriptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
```

Comment:
@ahamad-MS this parameter isn't necessary, but if you'd like to keep it, I would rename it to `-DomainResourceId`

Reply:
can you please provide more context here on why you think it is not needed and why to rename it to DomainResourceId. The reason I am asking is because we would like to have same experienece between Topic and Domain

Reply:
The reasoning here is pretty similar to the one outlined above for `-InputObject` vs. `-DomainObject` -- the `-ResourceId` isn't typically included for `New-*` cmdlets (since you shouldn't be able to pass a resource id for a resource you're trying to create). To specify that this is the resource id for the domain you're creating the key for, I would advise renaming this to `-DomainResourceId`.

Reply:
Changed.